onyo/lib/commands.py

```diff
@@ -1042,36 +1042,82 @@ def onyo_set(inventory: Inventory,
 
 @raise_on_inventory_state
 def onyo_tree(inventory: Inventory,
-              paths: Optional[list[Path]] = None) -> None:
+              dirs: Optional[list[tuple[str, Path]]] = None) -> None:
     """Print the directory tree of paths.
 
     Parameters
     ----------
     inventory: Inventory
         The inventory in which the directories to display are located.
 
-    paths: list of Path
-        The paths to directories for which to print the directory tree.
-        If no path is specified, `onyo_tree(inventory)` prints the
-        directory tree for the root of the inventory.
+    dirs: list tuple (str, Path)
+        A list of tuples containing a description and Path of a directory to
+        build a tree of.
+
+        The description is used as a text representation of what path the user
+        requested. This way, regardless of how the user requested a path
+        (relative, absolute, subdir, etc), it is always printed "correctly".
+
+        If no dirs are specified, ``onyo_tree(inventory)`` prints the directory
+        tree from the root of the inventory.
 
     Raises
     ------
     ValueError
         If paths are invalid.
     """
     # sanitize the paths
-    dirs = paths if paths else [inventory.root]
-    non_inventory_dirs = [str(p) for p in dirs if not inventory.repo.is_inventory_dir(p)]
+    dirs = dirs if dirs else [('.', inventory.root)]
```

Comment:
Yes. I felt that if we're implicitly operating from inventory root, then that's the scope of our world view. Precisely because it's implicit, it means the user doesn't imagine the outside world.

Including the FS path leading to the repo root would seem to run counter to that idea.

Reply:
Good catch, that this is a debatable spot. I wouldn't mind much, though:

1. Both - `'.'` and absolute path - are defensible options from the python interface POV, IMO. You know what `Inventory` object you are calling this on.
2. I don't really see a python interface usecase for this type of command to begin with, other than more sophisticated scripts that are meant to be used via CLI themselves. In which case you wouldn't really use the default anyway.

In short: I think it's fine either way.

Reply:
> I don't really see a python interface usecase for this type of command to begin with

I agree, and even less-so for using the python interface to generically list a tree of repo root without specifying the directory.

I actually think the right thing to do here is to remove the optional nature of the `dirs` argument. It removes the ambiguity of this situation, and allows me to remove the silencing of Pyre.

Reply:
I have made `dirs` required, and pushed.